include/rocksdb/options.h

```diff
@@ -1063,6 +1073,21 @@ struct ReadOptions {
   // Default: false
   bool ignore_range_deletions;
 
+  // 3 params below are added for differential snapshots
+  // If true iterator will return internal keys instead of user keys.
+  // Default: false
+  bool internal_keys;
+
+  // Iterator start timestamp; only SST files with
+  // creation_time >= iter_start_ts  will be scanned.
+  // Default: 0 (include all files)
+  uint64_t iter_start_ts;
```

Comment:
Isn't this doing the same thing as what cockroach db does with sst file filter? If so, why not use that approach?

Reply:
We can use the callback added by CockroachDB, it's just that this PR isn't merged yet (btw it can land now, right?), i can pick it and rebase on top yes.

Speaking on the reduction of properties..

 - We don't need iter_start_ts once the PR from CockroachDB guys lands
 - We don't really need to have a flag internal_keys, we can say that "if start sequence number is passed and is not zero, than return internal keys". A bit messy but documentation would help.
 - We do need to SeqNum filter, and it can't be covered by the callback from mentioned above, since it works at the different level, above individual file (table) iterators.